pkg/server/server.go

```diff
@@ -577,3 +583,45 @@ func setClusterDNSConfig(ctx context.Context, controlConfig *Config, configMap v
 	}
 	return nil
 }
+
+func setEncryptionHashAnnotation(ctx context.Context, nodes v1.NodeClient, controlConfig *config.Control) error {
```

Comment:
Can we move the hashing and such into a dedicated function that's called from `setControlPlaneRoleLabel`? Just pass it the controlConfig and node. 

Most of the logic around waiting for NODE_NAME, getting the node, adding annotations, etc is already in there, and copy-pasting it here it here (in addition to not being very DRY) will lead two twice the number of info-level logs spewing `Waiting for control-plane node agent startup` while the server's agent starts up.

Reply:
Reworked this function and `setControlPlaneRoleLabel` into a new single function `setLabelsAndAnnotations`